vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -129,12 +140,23 @@ public void handle(RoutingContext context) {
         path = indexPage;
       }
 
+      // Access fileSystem once here to be safe
+      FileSystem fs = fileSystem;
+      if (fs == null) {
+        if (classLoader == null) {
+          fs = context.vertx().fileSystem();
+        } else {
+          fs = ((VertxInternal)context.vertx()).fileSystem(classLoader);
```

Comment:
Does it create a new instance?
If yes, are we sure to not duplicate the worker pool used by the file system. If no, we may create a classloader leak, as we would need a way to clear the reference on the classloader.